src/OmniSharp.Roslyn.CSharp/Workers/Diagnostics/CSharpDiagnosticWorkerWithAnalyzers.cs

```diff
@@ -398,32 +409,23 @@ private void EmitDiagnostics(DocumentDiagnostics results)
             });
         }
 
-        public ImmutableArray<DocumentId> QueueDocumentsForDiagnostics()
-        {
-            var documentIds = _workspace.CurrentSolution.Projects.SelectMany(x => x.DocumentIds).ToImmutableArray();
-            QueueForAnalysis(documentIds, AnalyzerWorkType.Background);
-            return documentIds;
-        }
-
-        public async Task<ImmutableArray<DocumentDiagnostics>> GetAllDiagnosticsAsync()
+        public override async Task<ImmutableArray<DocumentDiagnostics>> GetAllDiagnosticsAsync()
         {
             var allDocumentsIds = _workspace.CurrentSolution.Projects.SelectMany(x => x.DocumentIds).ToImmutableArray();
             return await GetDiagnosticsByDocumentIds(allDocumentsIds, waitForDocuments: false);
```

Comment:
The CSharpDiagnosticWorker had the behavior that all work would be processed up front when All Diagnostics where requested. This restores that behavior by having it wait when analyzers are disabled.

```suggestion
            return await GetDiagnosticsByDocumentIds(allDocumentsIds, waitForDocuments: !AnalyzersEnabled);
```